client/asset/eth/eth.go

```diff
@@ -2402,18 +2536,379 @@ func (eth *ETHWallet) checkForNewBlocks(reportErr func(error)) {
+	if err != nil {
+		return nil, fmt.Errorf("failed to get best header: %w", err)
+	}
+	currentTip := hdr.Number.Int64()
+
+	confStatus := func(confs uint64, txHash common.Hash) *asset.ConfirmRedemptionStatus {
+		return &asset.ConfirmRedemptionStatus{
+			Confs:  confs,
+			Req:    txConfsNeededToConfirm,
+			CoinID: txHash[:],
+		}
+	}
+
+	resubmitTx := func(tx *types.Transaction, nonceOverride *uint64) (*asset.ConfirmRedemptionStatus, error) {
+		replacementTxHash, err := w.resubmitRedemption(tx, contractVer, nonceOverride, feeWallet)
+		if err != nil {
+			return nil, err
+		}
+
+		return confStatus(0, *replacementTxHash), nil
+	}
+
+	// Generally we rely on the transaction retrieved from geth, but we
+	// use our stored monitored tx for two reasons:
+	// - If the transaction was resubmitted, calls to ConfirmRedemption using
+	//   the old tx id will use the monitoredTxs map to determine the new one.
+	// - When geth cannot find the transaction, we can resubmit the redemption
+	//   using the tx stored in the monitoredTxs map.
+	w.monitoredTxsMtx.Lock()
+	monitoredTx, err := w.getLatestMonitoredTx(txHash)
+	if err != nil {
+		// We were unable to find the monitored tx, but we will attempt to continue
+		w.log.Error(err)
+	}
+	var blocksSinceSubmission int64
+	if monitoredTx != nil {
+		monitoredTx.mtx.Lock()
+		defer monitoredTx.mtx.Unlock()
+		if currentTip > monitoredTx.blockSubmitted {
+			blocksSinceSubmission = currentTip - monitoredTx.blockSubmitted
+		}
+		monitoredTxHash := monitoredTx.tx.Hash()
+		if monitoredTxHash != txHash {
+			w.log.Debugf("tx %s was replaced by %s since the last attempt to confirm redemption", txHash, monitoredTxHash)
+			txHash = monitoredTxHash
+		}
+	} else {
+		w.log.Warnf("tx %s is missing from monitored txs", txHash)
+	}
+	w.monitoredTxsMtx.Unlock()
+
+	gethTx, txBlock, err := w.node.getTransaction(w.ctx, txHash)
+	if errors.Is(err, asset.CoinNotFoundError) {
+		w.log.Errorf("ConfirmRedemption: geth could not find tx: %s", txHash)
+
+		if monitoredTx != nil && blocksSinceSubmission < blocksToWaitBeforeCoinNotFound {
+			return confStatus(0, txHash), nil
+		}
+
+		swapIsRedeemed, err := w.swapIsRedeemed(secretHash, contractVer)
+		if err != nil {
+			return nil, err
+		}
+		if swapIsRedeemed {
+			return confStatus(txConfsNeededToConfirm, txHash), nil
+		}
+
+		if monitoredTx != nil {
+			return resubmitTx(monitoredTx.tx, nil)
+		} else {
+			// If the tx is missing from both geth and the monitorTx, it is
+			// resubmitted individually.
+			txs, _, _, err := w.Redeem(&asset.RedeemForm{
+				Redemptions:   []*asset.Redemption{redemption},
+				FeeSuggestion: w.FeeRate(),
+			}, nil, nil)
+			if err != nil {
+				return nil, err
+			}
+			if len(txs) == 0 {
+				return nil, errors.New("no txs returned while resubmitting redemption")
+			}
+			var resubmittedTxHash common.Hash
+			copy(resubmittedTxHash[:], txs[0])
+			return confStatus(0, resubmittedTxHash), nil
+		}
+	} else if err != nil {
```

Comment:
As in previous comment, the `else` expresses that the prio `if`  block might _not_ return, but it does.  To clearly express your the intent, remove the `else`.